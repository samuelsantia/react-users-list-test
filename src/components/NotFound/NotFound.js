import React from 'react';
import styled from 'styled-components/macro';
import Typography from '../Typography';

export const NotFoundWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  height: 80%; /* not 100% vertical center */

  h1 {
    --font-increase: 10;
    --rhythm-increase: 3;
    color: var(--color-red);
  }
`;

function NotFound() {
  return (
    <NotFoundWrapper>
      <div>
        <Typography variant="h1" weight="bold">
          404
        </Typography>
        <Typography>Page not found</Typography>
      </div>
    </NotFoundWrapper>
  );
}

export default NotFound;
