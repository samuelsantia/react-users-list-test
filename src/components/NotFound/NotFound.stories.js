import React from 'react';
import styled from 'styled-components/macro';

import NotFound from './NotFound';

const StoryWrapper = styled.div`
  height: 50vh;
  z-index: -1;
`;

export default {
  title: 'NotFound',
  component: NotFound,
  decorators: [(storyFn) => <StoryWrapper>{storyFn()}</StoryWrapper>],
};

export const basic = () => <NotFound />;

export const withCustomText = () => <NotFound>Custom not found text</NotFound>;
