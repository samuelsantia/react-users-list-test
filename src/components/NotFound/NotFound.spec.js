import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import NotFound from './NotFound';

describe('<NotFound />', () => {
  it('should match snapshot', () => {
    const { container } = render(<NotFound />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
