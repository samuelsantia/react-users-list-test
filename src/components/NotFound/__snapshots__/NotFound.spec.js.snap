// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> should match snapshot 1`] = `
.c1 {
  --font-increase: var(--typo-h1-increase,5);
  --rhythm-increase: var(--typo-h1-rhythm,2);
  font-weight: var(--weight-bold,bold);
}

.c2 {
  --font-increase: var(--typo-body1-increase,0);
  --rhythm-increase: var(--typo-body1-rhythm,0);
  font-weight: var(--weight-normal,normal);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  height: 80%;
}

.c0 h1 {
  --font-increase: 10;
  --rhythm-increase: 3;
  color: var(--color-red);
}

<div
  class="c0"
>
  <div>
    <h1
      class="c1"
    >
      404
    </h1>
    <p
      class="c2"
    >
      Page not found
    </p>
  </div>
</div>
`;
